Generals/Code/GameEngine/Source/GameClient/Eva.cpp

```diff
@@ -213,7 +214,7 @@ void Eva::update()
 		return;
 	}
 
-	m_localPlayer = ThePlayerList->getLocalPlayer();
+	m_localPlayer = TheControlBar->getCurrentlyViewedPlayer();
```

Comment:
So this can be NULL now? Can this cause crashing?

Reply:
How about move the null test from `messageShouldPlay` to `update`? Looks to be more efficient that way.

Reply:
I did consider that, but was unsure if messages could be left in the pipe / durations not updated if the player stops observing another player.